tools/isobuild/compiler-plugin.js

```diff
@@ -179,17 +179,35 @@ export class CompilerPluginProcessor {
         buildmessage.enterJob({
           title: jobTitle
         }, function () {
-          var inputFiles = _.map(resourceSlots, function (resourceSlot) {
-            return new InputFile(resourceSlot);
-          });
+          let processOne = sourceProcessor.userPlugin.processOneFileForTarget;
+          if (typeof processOne === "function") {
+            // Instead of running the processOneFileForTarget method now,
+            // call the addLazyJavaScript method of the InputFile so that
+            // compilation can be performed when/if necessary. Compiler
+            // plugins can opt into this behavior by defining a method
+            // called processOneFileForTarget, which indicates they are
+            // willing to compile individual files, instead of a complete
+            // list of files all at once (processFilesForTarget).
+            resourceSlots.forEach(slot => {
+              new InputFile(slot).addLazyJavaScript(buildmessage.markBoundary(
```

Comment:
Concern: we don't actually know at this point that the compiler plugin would have called `addJavaScript` with the result of `processOneFileForTarget`, unless we assume that `processOneFileForTarget` is only used for JS files.